.github/workflows/push.yml

```diff
@@ -0,0 +1,55 @@
+name: Push - Matrix Tests
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  test-on-os-node-matrix:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest] # due to node-gyp & node compatibility issues, windows 2022 won't work for all node versions
```

Comment:
Is this comment still needed?

Can we come up with a more self-documenting name than `push` as other GHAs also run on `git push`.

Reply:
> Can we come up with a more self-documenting name than `push` as other GHAs also run on `git push`.

Yep. 

Given the new `ci` action, I removed the redundant run of tests and renamed the leftover for what they are (S3 Tests).

Once the bucket is created (note: Object Ownership -> ACLs enabled) and repo secrets are set this works as expected (see: https://github.com/ronilan/node-pre-gyp/actions/runs/9722621262)